cli/core/checkpoint.go

```diff
@@ -10,10 +10,69 @@ import (
 	"strconv"
 
 	eigenpodproofs "github.com/Layr-Labs/eigenpod-proofs-generation"
+	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/core/onchain"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/fatih/color"
 )
 
+const PROOFS_PER_BATCH = 60 // about 60 proofs fit per transaction.
+
+func SubmitCheckpointProof(ctx context.Context, owner, eigenpodAddress string, chainId *big.Int, proof *eigenpodproofs.VerifyCheckpointProofsCallParams, eth *ethclient.Client, batchSize uint64, noPrompt bool) ([]*types.Transaction, error) {
+	allProofChunks := chunk(proof.BalanceProofs, PROOFS_PER_BATCH)
+
+	transactions := []*types.Transaction{}
+	if !noPrompt {
+		PanicIfNoConsent(fmt.Sprintf("This will call EigenPod.VerifyCheckpointProofs() %d time(s), to complete your checkpoint.", len(allProofChunks)))
+	}
+
+	color.Green("calling EigenPod.VerifyCheckpointProofs() (using %d txn(s), max(%d) proofs per txn)", len(allProofChunks), batchSize)
+
+	for i := 0; i < len(allProofChunks); i++ {
+		balanceProofs := allProofChunks[i]
+		txn, err := SubmitCheckpointProofBatch(owner, eigenpodAddress, chainId, proof, balanceProofs, eth, batchSize)
+		if err != nil {
+			// failed to submit batch.
+			return transactions, err
+		}
+		transactions = append(transactions, txn)
+		fmt.Printf("Submitted chunk %d/%d -- waiting for transaction...: ", i+1, len(allProofChunks))
+		bind.WaitMined(ctx, eth, txn)
```

Comment:
submitting in parallel is prob the move here. 50 consecutive txs and such will be a pain

Reply:
separate pr but agreed